Sources/Receipt/ListReceiptViewPresenter.swift

```diff
@@ -29,25 +29,49 @@ final class ListReceiptViewPresenter {
     private unowned let view: ListReceiptView
 
     private var offset = 0
-    private let limit = 20
+    private let userReceiptLimit = 20
+    private var prepaidCardToken: String?
+    private let prepaidCardReceiptLimit = 20
+    private var prepaidCardReceiptCreatedAfter = Calendar.current.date(byAdding: .year, value: -1, to: Date())
 
     private var isLoadInProgress = false
     private(set) var areAllReceiptsLoaded = true
     private(set) var sectionData = [(key: Date, value: [HyperwalletReceipt])]()
 
     /// Initialize ListReceiptPresenter
-    init(view: ListReceiptView) {
+    init(view: ListReceiptView, prepaidCardToken: String? = nil) {
         self.view = view
+        self.prepaidCardToken = prepaidCardToken
     }
 
-    func listReceipt() {
+    func listReceipts() {
```

Comment:
and here add a switch operator
```
        switch listReceiptType {
        case .user:
            listUserReceipts()

        case let .prepaidCard(prepaidCardToken):
            listPrepaidCardReceipts(prepaidCardToken)
        }
```
as well as for account receipts in the future:
```
        case let .account(programToken, accountToken):
            listAccountReceipts(programToken, accountToken)
```

Reply:
Please see the above explanation